Feature/header

bzgaspar opened the pull request:
Create UI of the Header. Fetch categories from contentful and display as nav items. Responsive from mobile to desktop

[Preview](https://audiophile-e-commerce-website-git-feature-header-itsmenek.vercel.app?_vercel_share=ACWuondWozW40RZCceox6abK6LjYAjpG)

bzgaspar committed:
feat: Header

Create UI of the Header. Fetch categories from contentful and display as nav items. Responsive from mobile to desktop

eremocal commented on src/app/page.tsx:
```diff
@@ -1,3 +1,3 @@
-export default function Home() {
-  return <main>page</main>
+export default async function Home() {
```
any reason why this is async?

eremocal commented on src/contentful/categories.ts:
```diff
@@ -0,0 +1,18 @@
+import contentfulClient from './contentful'
+import { parseThumbnail } from './helpers/parseThumbnail'
+
+export async function parseContentfulCategories() {
```
I think it might be better to rename this to `getContentfulCategories` or `getCategories`

eremocal commented on src/contentful/contentful.ts:
```diff
@@ -0,0 +1,25 @@
+  const isPreviewMode = VERCEL_ENV === 'preview'
+
+  const params: CreateClientParams & { accessToken: string } = {
+    space: CONTENTFUL_SPACE_ID!,
```
any reason why do we have a `!`

eremocal commented on src/contentful/contentful.ts:
```diff
@@ -0,0 +1,25 @@
+    space: CONTENTFUL_SPACE_ID!,
+    environment: 'master',
+    accessToken: isPreviewMode
+      ? CONTENTFUL_PREVIEW_ACCESS_TOKEN!
```
same with here

bzgaspar committed:
refactor: Refactor

Remove async in page. Remove '!' in space id & token.

bzgaspar committed:
refactor: refactor

Move getContenfulCategories in services folder. Move helpers to src.

feat: Add Top Nav and Categories

refactor: Refactor icon for menu & close

feat: Add CartModal.

Create mock data. Create Cart modal. Separate ovelay in another file.

eremocal commented on src/contentful/contentfulClient.ts:
```diff
@@ -0,0 +1,30 @@
+
+  const params: CreateClientParams & { accessToken: string } = {
+    space: space_id,
+    environment: 'master',
```
could you please add this on env as well?

bzgaspar committed:
fix deployment error

bzgaspar committed:
fix: Overlay error. Move CONTENTFUL_ENVIRONMENT

bzgaspar committed:
fix: warning in flex-start

bzgaspar committed:
refactor: remove variable in contentful. change env

eremocal commented on src/contentful/contentfulClient.ts:
```diff
@@ -3,25 +3,19 @@ const {
   CONTENTFUL_ACCESS_TOKEN,
   CONTENTFUL_PREVIEW_ACCESS_TOKEN,
-  CONTENTFUL_ENVIRONMENT,
+  ENV,
```
please prefix this to `CONTENTFUL_` eg: `CONTENTFUL_ENVIRONMENT`

bzgaspar committed:
Refactor: change ENV to CONTENTFUL_ENV

bzgaspar committed:
refactor contentful environment
